apps/web/components/Modal.tsx

```diff
@@ -0,0 +1,51 @@
+"use client";
+import { ReactNode, useEffect, useRef } from "react";
+import { XMarkIcon } from "@heroicons/react/24/outline";
+
+interface ModalProps {
+  icon?: ReactNode;
+  title?: string;
+  onClose: () => void;
+  children: ReactNode;
+  isOpen: boolean;
+}
+
+export function Modal({ icon, title, onClose, children, isOpen }: ModalProps) {
+  const dialogRef = useRef<HTMLDialogElement>(null);
+
+  useEffect(() => {
+    if (isOpen) {
+      dialogRef.current?.showModal();
+    } else {
+      dialogRef.current?.close();
+    }
+  }, [isOpen]);
+
+  const handleClose = () => {
+    onClose();
+    dialogRef.current?.close();
```

Comment:
Dont think we should close it since we already close from outside through the isOpen prop